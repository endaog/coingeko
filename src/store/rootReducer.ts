import { combineReducers } from "redux"

import coinsReducer from "../redux/coins/coins.reducer"

const rootReducer = combineReducers({
  coins: coinsReducer,
})

export type AppState = ReturnType<typeof rootReducer>

export default rootReducer
