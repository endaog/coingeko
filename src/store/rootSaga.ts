import { all, fork } from "redux-saga/effects"

import coinsSaga from "../redux/coins/coins.saga"

export function* rootSaga() {
  yield all([fork(coinsSaga)])
}
