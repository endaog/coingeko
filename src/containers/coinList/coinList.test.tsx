import React from "react"
import { render } from "@testing-library/react"
import { Provider } from "react-redux"
import { CoinList } from "."
import { BrowserRouter } from "react-router-dom"
import store from "../../store"

const spyScrollTo = jest.fn()

describe("Coin list", () => {
  beforeEach(() => {
    Object.defineProperty(global.window, "scrollTo", { value: spyScrollTo })
    spyScrollTo.mockClear()
  })

  test("renders the loader when no content exists", () => {
    const testStore = store
    const { asFragment, getByText } = render(
      <Provider store={testStore}>
        <BrowserRouter>
          <CoinList />
        </BrowserRouter>
      </Provider>
    )
    // Check loading message present
    expect(getByText(/Loading.../i)).toBeInTheDocument()

    // Take a snapshot for regressions
    expect(asFragment()).toMatchSnapshot()

    // Scroll to top
    expect(spyScrollTo).toHaveBeenCalledWith({
      top: 0,
      left: 0,
      behavior: "smooth",
    })
  })
})
