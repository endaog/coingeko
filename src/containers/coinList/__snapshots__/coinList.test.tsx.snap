// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coin list renders the loader when no content exists 1`] = `
<DocumentFragment>
  <section
    class="main__container"
  >
    <div
      class="loader"
    >
      Loading...
    </div>
  </section>
</DocumentFragment>
`;
